File: java-to-kotlin-starter-guide-master/src/main/java/com/lannstark/lec12/Lec12Main.java
package com.lannstark.lec12;

import com.lannstark.lec12.kotlin.Person;

public class Lec12Main {

  public static void main(String[] args) {
    //Java에서 Kotlin companion object를 사용하려면 3가지 방법이 있다.
    // 1. companion object에 이름이 없는 경우 - Companion을 직접 명시
//    Person.Companion.newBaby("Solar");
    // 2.  companion object에 이름이 없는 경우 - companion object에 @JvmStatic 을 붙여준다.
    Person.newBaby("Solar");
    // 3. companion object에 이름이 있는 경우
    Person.Factory.newBaby("Solar");
  }
}

File: java-to-kotlin-starter-guide-master/src/main/java/com/lannstark/lec12/kotlin/Lec12Main.kt
package com.lannstark.lec12.kotlin

import com.lannstark.lec12.Movable

fun main() {
    //익명클래스 사용
    moveSomething(object : Movable {
        override fun move() {
            println("무브 무브")
        }

        override fun fly() {
            println("날다 날다")
        }
    })
}

private fun moveSomething(movable: Movable) {
    movable.move()
    movable.fly()
}
